util/topgen/c.py

```diff
@@ -95,6 +95,9 @@ def __init__(self, top_info):
         self.top = top_info
         self._top_name = Name(["top", top_info["name"]])
 
+        # The .c file needs the .h file's relative path, store it here
+        self.header_path = ""
```

Comment:
Can this be `None` if unset? That way, it actually has a different type from what it would have with a valid value.

Reply:
I've changed this, but I hate None creeping in everywhere when before it was a string and always a string.